app/services/one_off/fhir/prescription_drug_exporter.rb

```diff
@@ -0,0 +1,118 @@
+require "fhir_models"
+
+module OneOff
+  module Fhir
+    class PrescriptionDrugExporter
+      attr_reader :prescription_drug
+
+      def initialize(prescription_drug)
+        @prescription_drug = prescription_drug
+      end
+
+      def export
+        FHIR::MedicationRequest.new(
+          contained: [
+            FHIR::Medication.new(
+              identifier: FHIR::Identifier.new(
+                value: prescription_drug.id
+              ),
+              code: FHIR::CodeableConcept.new(
+                coding: FHIR::Coding.new(
+                  system: "http://www.nlm.nih.gov/research/umls/rxnorm",
+                  code: prescription_drug.rxnorm_code,
+                  display: prescription_drug.name
+                )
+              )
+            )
+          ],
+          status: prescription_drug.is_deleted ? "ended" : "active",
+          intent: "proposal",
+          medicationReference: FHIR::Reference.new(
+            id: FHIR::Medication.new(
+              identifier: FHIR::Identifier.new(
+                value: prescription_drug.id
+              ),
+              code: FHIR::CodeableConcept.new(
+                coding: FHIR::Coding.new(
+                  system: "http://www.nlm.nih.gov/research/umls/rxnorm",
+                  code: prescription_drug.rxnorm_code,
+                  display: prescription_drug.name
+                )
+              )
+            )
```

Comment:
We shouldn't have to repeat everything here (not sure if this would lead to valid FHIR). We can directly reference the contained resource like so:

```ruby
FHIR::Reference.new(reference: "##{contained_thing_id}")
```

Reply:
Going to go with your example, since it's closer to the examples

Reply:
I guess if the reference is to a contained resource, it's all the same

Reply:
Uh, read about contained resources. `contained` + `medicationReference` makes a lot of sense now and seems better to do than exporting prescription drugs separately. There can be a lot of duplicates, and it's not like just exporting protocol drugs. It's also not how we store things in Simple.